### ISSUE TITLE
linter: False Positive: `eslint-plugin-vitest(prefer-import-in-mock)` triggers on test.each labels

### ISSUE THREAD

**Issue by antoinezanardi** [oxc-project/oxc#19566]
### What version of Oxlint are you using?

1.49.0

### What command did you run?

oxlint --type-aware -c configs/oxlint/oxlint.config.jsonc

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
// Replace this with your actual config file contents.
// It can be helpful to keep it minimal, if possible, but please ensure that it actually reproduces the issue you're reporting.
{
  "rules": {
    "eslint-plugin-vitest/prefer-import-in-mock": "error"
  }
}
```

### Describe the bug

The rule `eslint-plugin-vitest(prefer-import-in-mock)` incorrectly flags the test name string in `it.each` / `test.each` (and likely `describe.each`) calls.

The linter seems to misidentify the test description string (e.g., `"$test"`) as a module path inside a `vi.mock()` call, suggesting to replace it with a dynamic import (`import('$test')`). This is invalid for test descriptions.

### Input code

```typescript
import { describe, it, expect } from 'vitest';

const validAppMetadataDto = { version: 1 };
const APP_METADATA_DTO = { parse: (obj: any) => obj };

describe('reproduction', () => {
  it.each([
    { test: 'should fail', value: 'invalid' },
    { test: 'should pass', value: 1 },
  ])("$test", ({ value }) => {
     // The string "$test" above triggers the false positive
    const invalidDto = Object.assign({}, validAppMetadataDto, { version: value });
    expect(invalidDto).toBeDefined();
  });
});
```

### Actual output

```text
  × eslint-plugin-vitest(prefer-import-in-mock): Mocked modules must be dynamic imported.
    ╭─[src/reproduction.spec.ts:10:8]
  9 │     { test: 'should pass', value: 1 },
 10 │ ╭─▶   ])("$test", ({ value }) => {
 11 │ │      // The string "$test" above triggers the false positive
 12 │ │     const invalidDto = Object.assign({}, validAppMetadataDto, { version: value });
 13 │ │     expect(invalidDto).toBeDefined();
 14 │ ╰─▶   });
 15 │     
    ╰────
  help: Dynamic import improves the type information and IntelliSense. Substitute `$test` with `import('$test')`
```

**Comment by camc314** [comment]
https://github.com/oxc-project/oxc/pull/19555
